From 07232d126c5364100b61232de1ff752d3214a0f1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 25 Jul 2024 18:53:59 +0300
Subject: [PATCH] [FSW-7134] pldm_bloader: use correct emb specs

pldm_bloader application was including headers from EMB specs folder.
But for fs1 it should take the headers from emb specs/fs folder.
To achieve that, the hardcoded path is removed from the makefile,
and the platform specific makefile (that holds the correct path) is
included.

In a addition, pif_ts module is excluded from the build as it's not
relevant for the pldm_bloader nor for FS1.

Change-Id: I0120d5596fdc8da58f3c3adc1f0916584c724b4a
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pldm_bloader/Makefile b/app/pldm_bloader/Makefile
index c8ce6e3..6f84400 100644
--- a/app/pldm_bloader/Makefile
+++ b/app/pldm_bloader/Makefile
@@ -8,9 +8,9 @@
 
 EMB_COMMON_MOD_DIR := ${PREBOOT_ROOT_DIR}/emb_common_mod
 
+include $(HL_DEV_ARCH)/Makefile
+
 CFLAGS += -I$(SPECS_DIR)
-CFLAGS += -I$(SPECS_EMBEDDED_DIR)
-CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
@@ -31,6 +31,7 @@
 
 PLDM_BLOADER_SRCS_C := $(shell ls *.c)
 PLDM_BLOADER_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/common/*.c)
+PLDM_BLOADER_SRCS_C := $(filter-out $(PREBOOT_ROOT_DIR)/lib/common/pif_ts.c,$(PLDM_BLOADER_SRCS_C))
 PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/common.c)
 PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
 PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
